Source/WebCore/css/CSSCursorImageValue.cpp

```diff
@@ -61,23 +62,33 @@ CSSCursorImageValue::~CSSCursorImageValue() = default;
 
 String CSSCursorImageValue::customCSSText() const
 {
-    auto text = m_imageValue.get().cssText();
+    auto text = m_imageValue->cssText();
     if (!m_hotSpot)
         return text;
-    return makeString(text, ' ', m_hotSpot->x(), ' ', m_hotSpot->y());
+    return makeString(text, ' ', m_hotSpot->first().cssText(), ' ', m_hotSpot->second().cssText());
 }
 
 bool CSSCursorImageValue::equals(const CSSCursorImageValue& other) const
 {
-    return m_hotSpot == other.m_hotSpot && compareCSSValue(m_imageValue, other.m_imageValue);
+    return compareCSSValue(m_imageValue, other.m_imageValue)
+        && compareCSSValuePtr(m_hotSpot, other.m_hotSpot);
 }
 
-RefPtr<StyleImage> CSSCursorImageValue::createStyleImage(Style::BuilderState& state) const
+RefPtr<StyleCursorImage> CSSCursorImageValue::createStyleImage(Style::BuilderState& state) const
 {
     auto styleImage = state.createStyleImage(m_imageValue.get());
     if (!styleImage)
         return nullptr;
-    return StyleCursorImage::create(styleImage.releaseNonNull(), m_hotSpot, m_originalURL, m_loadedFromOpaqueSource);
+
+    std::optional<IntPoint> hotSpot;
+    if (m_hotSpot) {
+        // FIXME: Should we clamp or round instead of just casting from double to int?
```

Comment:
Sure would be neat to fix this as a drive-by if we can figure out what is the best behavior.

Reply:
Yeah, seems like rounding would be ideal, but I wanted to do that separately just incase it breaks something and needs to be rolled out.

Reply:
To be fair, it’s not entirely trivial because I realize there are 3 orthogonal small questions:

1) Should we round instead of truncating?
2) Should we clamp to the integer range instead of accepting the behavior that casting an out of range value from double to int does (can’t remember what that is)?
3) Should we clamp further based on rules about what a valid hot spot is? Apparently this is a WebKit difference from the other browser engines: https://bugs.webkit.org/show_bug.cgi?id=102398

But I still think these are things you could handle if you are motivated enough.